types/virtual-dom/index.d.ts

```diff
@@ -42,13 +42,21 @@ declare namespace VirtualDOM {
     type: string; // 'VirtualNode'
   }
 
+  interface VNodeConstructor {
+    new (tagName: string, properties: VProperties, children: VTree[], key?: string, namespace?: string): VNode;
+  }
+
   interface VText {
     text: string;
     new(text: any): VText;
```

Comment:
To my understanding, this existing line is wrong and should be removed (the `VTextContructor` being added here is the correct typing). This line makes it possible to write things like
```ts
const text = new VText("Hello");
const why = new text("World");
```
which is nonsense. This PR would be a good chance to fix up the old types.

The same could be done with the `VPatch` definition, but that's fine to leave for another PR.

Reply:
@dphilipson turns out `patch` doesn't have a new operator according to the library (it's just a function), so updated the typings accordingly

Reply:
yeah I looked through the virtual-dom library and `patch` is used as a function and not as a constructor, so I just removed the `new` constructor.